Pequenas correcoes, eventos do Meetup, Mini-Biggy

andrecalil opened the pull request:
- Corrigi o link do favicon e ie10mobile.css que estavam quebrados
- Adicionei um novo repositório (ExternalRepository) que carrega as informações de fora. Por enquanto, apenas os eventos. As empresas vou carregar de um arquivo JSON mantido no Github, ainda por fazer.
- Novo link para todos os eventos (link geral do grupo no Meetup)
- Mini-biggy já em uso, fazendo deserialização da requisição do Meetup
#7 #9

andrecalil committed:
Merge pull request #2 from dotnetfloripa/master

Update from origin

andrecalil committed:
Pequenas correcoes, eventos do Meetup, Mini-Biggy

Corrigi o link do favicon e ie10mobile.css que estavam quebrados.
Adicionei um novo repositório (ExternalRepository) que carrega
informações de fora. Por enquanto, apenas os eventos. As empresas vou
carregar de um arquivo JSON mantido no Github, ainda por fazer.  #7 #9

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,105 @@
+    {
+        public ExternalRepository()
+        {
+            #region Events from Meetup
```
no fucking way

andrecalil replied:
Que feedback dahora. Qual o problema?

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,105 @@
+                {
+                    string serializedEvents = response.Content.ReadAsStringAsync().Result;
+
+                    var jsonSerializer = new MiniBiggy.Serializers.JsonSerializer();
```
?? usa o json.net pra deseralizar. pode tirar ro minibiggy do projeto por enquanto

andrecalil replied:
Seria outra dependência. Como já vamos usar o minibiggy mesmo, preferi reduzir o número de dependências.

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,105 @@
+
+            try
+            {
+                var httpClient = new HttpClient();
```
porque isso tá no construtor?

andrecalil replied:
Segui o padrão do outro. Não precisaria.

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,105 @@
+        public Company[] Companies { get; }
+        public Job[] Jobs { get; }
+
+        public IQueryable<Event> GetEvents() => Events.AsQueryable();
```
porquê a real lógica dos métodos não está dentro dos métodos?

andrecalil replied:
Poderia estar. Ainda quero adicionar uma lógica de cache e transformar esse serviço em singleton, assim evitamos carregar em todas as requisições.

andrecalil committed:
ExternalRepo revisado, mais fontes externas

andrecalil commented:
@luisrudge organizei melhor o ExternalRepo, dá uma olhada. Só procura fazer comentários melhores do que aquele primeiro, além de não muito claro foi meio grosseiro.

@andrecarlucci minibiggy em ação! Fiz uma PersistentList específica para carregar empresas e vagas, dá uma olhada.

andrecalil commented:
Mais detalhes:
- Eventos: estão vindo do Meetup, pela API deles.
- Empresas: arquivo JSON mantido no Github.
- Vagas: arquivo JSON mantido no Github.

Os arquivos estão em https://github.com/andrecalil/dotnetfloripa-content, se adotarmos essa fonte seria legal criar outro repo sob /dotnetfloripa, pra ficar mais organizado.
Ainda não fiz a página de vagas, precisaremos conversar melhor sobre isso. Por enquanto, só deixei pronta a recuperação.

O ExternalRepository está configurado como Singleton. Há um cache de 24hs para cada tipo de objeto. Desta forma, para uma mesma instância do site, as informações não serão carregadas o tempo todo (o que deixaria a navegação lenta).

luisrudge commented:
@andrecalil cara, me desculpa! eu deveria ter sido mais claro e menos cuzão. deveria só ter falado pra tirar as regions. sorry! ❤️
Vou revisar novamente o PR

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -13,93 +16,121 @@ public class ExternalRepository : IAppRepository
+                        string serializedEvents = response.Content.ReadAsStringAsync().Result;
 
-            #endregion
+                        var jsonSerializer = new MiniBiggy.Serializers.JsonSerializer();
```
se a deserialização é essencial pro nosso app, não podemos depender do serializador do mini biggy. Usa o json.net direto.

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -13,93 +16,121 @@ public class ExternalRepository : IAppRepository
+            {
+                try
+                {
+                    var companiesBiggy = new PersistentList<Company>(new GitHubDataStore("andrecalil/dotnetfloripa-content/master/companies.json"), new JsonSerializer(), new SaveOnlyWhenRequested());
```
ainda não entendi o que o minibiggy tá agregando de benefício aqui.. porque não deixar os arquivos .json no mesmo projeto e ler o arquivo quando a aplicação iniciar?

andrecalil replied:
O minibiggy não seria _realmente_ necessário. A mudança que estou trazendo é ter os **dados em uma fonte externa**, assim o projeto não precisa ser republicado a cada alteração. Inclusive por isso que sugeri um repo diferente para ter os arquivos de empresas e vagas (pq, se fosse no mesmo repo, seria republicado pelo CD).

Eu poderia ler os arquivos sem usar o minibiggy para nada, mas estou atendendo ao pedido do @andrecarlucci no #6.

Sobre serialização, o minibiggy usa o json.net. De novo: poderíamos ir direto, mas estamos ajudando a testar o projeto.

luisrudge replied:
ACho legal testar o mini biggy pra validar o framework, mas a gente só tá lendo dados.. Se a gente fosse ter leitura e escrita até acharia legal, mas não é o caso.. estamos só lendo arquivos da internet (no nosso caso, do github). Por isso achei um pouco demais o mini biggy.

Eu não vejo problema em republicar o app toda vez que alguém adicionar uma vaga ou empresa. O processo é todo automatizado, então, por mim, pode publicar todo dia. E ainda estimula as pessoas a fazerem alterações no nosso repositório principal, o que é legal.

Usando a serialização do mini biggy, você tá acoplando o nosso app ao mini biggy pra sempre. Esse acoplamento não pode ter.

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -13,93 +16,121 @@ public class ExternalRepository : IAppRepository
+        }
 
-            List<Event> events = new List<Event>();
+        private readonly TimeSpan ReloadThreshold = new TimeSpan(24, 0, 0);
```
ao invés de fazer o controle do cache manualmente, você pode usar o recurso de cache do proprio aspnet https://docs.asp.net/en/latest/performance/caching/memory.html

andrecalil committed:
NO MOAR MINIBIGGY

andrecalil commented:
Removi o minibiggy (sorry @andrecarlucci) e organizei o ExternalRepository um pouco melhor.

Acho que ter os dados em fonte externa e em um repo próprio facilita _muito_ para engajarmos a comunidade a mandar seus PRs para publicações de vagas.

andrecalil commented:
@luisrudge Travis deu erro pq não conseguiu clonar o repo. Sei que tá rolando um problema sério de backbones lá pra cima, pode ter relação.

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,144 @@
+                if (response.IsSuccessStatusCode)
+                    result = response.Content.ReadAsStringAsync().Result;
+            }
+            catch (Exception)
```
acho legal relançar a exceção aqui

andrecalil replied:
Mas daí vai dar pau no site se por algum motivo o arquivo não for lido. É isso mesmo?

andrecalil replied:
Esquece, tem o try no outro nível.

luisrudge replied:
👍

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,144 @@
+
+                HttpResponseMessage response = httpClient.GetAsync(resourceURI).Result;
+
+                if (response.IsSuccessStatusCode)
```
ao invés de fazer esse if, usa o `response.EnsureSuccessStatusCode()` e deixa a exceção acontecer

andrecalil replied:
Nice

luisrudge commented on DotNetFloripa/DotNetFloripa.DataCore/ExternalRepository.cs:
```diff
@@ -0,0 +1,144 @@
+            }
+        }
+
+        private void LoadCompanies ()
```
tem um espaço a mais

andrecalil replied:
Tem          memo

luisrudge commented:
Tá bem melhor agora 🎉 
E concordo com o seu ponto sobre os arquivos externos.

andrecalil committed:
Corressõnx para o champs

Tirei o try/catch do primeiro nível pq ele ficaria absolutamente inútil.
Se um dia adicionarmos logs (tipo um NewRelic ou Raygun da vida), daí
botamos de volta.

andrecalil commented:
> fatal: unable to access 'https://github.com/dotnetfloripa/site.git/': Could not resolve host: github.com

Oh joy

luisrudge commented:
É.. tá rolando um ddos tenso

luisrudge commented:
Cara, antes de continuar.. manda um PR aqui: https://github.com/dotnetfloripa/site-data pra gente popular os dados (e aponta o código pra lá também). Valeu!

andrecalil committed:
Repo do grupo como fonte de dados

luisrudge commented:
acabei de testar aqui, tá ótimo! acho que pode deletar o antigo `AppRepository` e já era.
Valeu!

andrecalil committed:
Remocao do repositorio hard coded

luisrudge commented:
feito! valeu @andrecalil
